### ISSUE TITLE
Measure IL is tricky

### ISSUE THREAD

**Issue by spouliot** [rolfbjarne/dotnet-appsize-report#1]
That looks like a nice tool 😄 

You already know it but (for the rest of the world) measuring IL can be tricky, even more for iOS where it's AOT'ed and stripped. IOW the tool measure something that is not even part of the final app so, at best, YMMV 😉

Many methods (e.g. properties) _managed_ size are larger from their metadata (methoddef table includes method name, tables for all custom attributes) than the IL itself (e.g. `ldc.i4.0 + ret`).

If the linker/trimmer can inline such methods (**and remove** original code/metadata [1]) then there are interesting savings (from metadata) in app size to realize.

That could either be reported separately by the tool...

or the IL size could be renamed to "Member size" and approximate the metadata size (based on the data in the tables).

[1] like it was done on the legacy SDK

**Comment by rolfbjarne** [comment]
> IOW the tool measure something that is not even part of the final app so, at best, YMMV

Yeah, it's mostly meant as a tool to figure out what the linker didn't remove (and maybe even why it didn't remove it), with some sort of weighing based on a very incomplete calculation of the IL Size.

> If the linker/trimmer can inline such methods (**and remove** original code/metadata [1]) then there are interesting savings (from metadata) in app size to realize.

Amazingly this gets tricky, because calling a property can have side effects - in particular if the class has a static constructor, that static constructor might not be executed if a property is inlined elsewhere.

And yes, the BCL depends on this behavior in at least one scenario...
